library/core/src/androidTest/java/com/google/android/exoplayer2/upstream/ContentDataSourceTest.java

```diff
@@ -0,0 +1,40 @@
+package com.google.android.exoplayer2.upstream;
+
+import android.content.ContentResolver;
+import android.content.Context;
+import android.net.Uri;
+import android.test.InstrumentationTestCase;
+
+import static com.google.android.exoplayer2.upstream.AndroidDataSourceConstants.NULL_DESCRIPTOR_URI;
+import static com.google.android.exoplayer2.upstream.AndroidDataSourceConstants.SAMPLE_MP4_BYTES;
+import static com.google.android.exoplayer2.upstream.AndroidDataSourceConstants.SAMPLE_MP4_PATH;
+import static com.google.android.exoplayer2.upstream.AndroidDataSourceConstants.TEST_DATA_PROVIDER_AUTHORITY;
+
+public class ContentDataSourceTest extends InstrumentationTestCase {
+
+  public void testValidContentDataSource() throws Exception {
+    Context context = getInstrumentation().getContext();
+    ContentDataSource dataSource = new ContentDataSource(context);
+    Uri contentUri = new Uri.Builder()
+            .scheme(ContentResolver.SCHEME_CONTENT)
+            .authority(TEST_DATA_PROVIDER_AUTHORITY)
+            .path(SAMPLE_MP4_PATH).build();
+    DataSpec dataSpec = new DataSpec(contentUri);
+    long sourceLengthBytes = dataSource.open(dataSpec);
+
+    assertEquals(SAMPLE_MP4_BYTES, sourceLengthBytes);
```

Comment:
Hm, I tried to modify this test to also validate that the data being read was correct, but it appears the data read from the DataSource is not as expected. Any ideas?